src/cuda-sha256/sha256.cu

```diff
@@ -194,7 +183,7 @@ static int sha256_compress(hash_state * md, const unsigned char *buf)
    @param md   The hash state you wish to initialize
    @return CRYPT_OK if successful
 */
-int __host__ __device__ sha256_init(hash_state * md)
+inline int __host__ __device__ sha256_init(hash_state * md)
```

Comment:
Don't believe this inline will do anything. do you see an effect?

Reply:
Had to move the definition to the .h file

Reply:
Right. So all the sha256 functions should be like this. Hence just including the .cu in the compilation unit that you want it in.